session/session.go

```diff
@@ -475,7 +475,15 @@ func (s *session) doCommitWithRetry(ctx context.Context) error {
 			txnSizeRate := float64(txnSize) / float64(kv.TxnTotalSizeLimit)
 			maxRetryCount := commitRetryLimit - int64(float64(commitRetryLimit-1)*txnSizeRate)
 			err = s.retry(ctx, uint(maxRetryCount))
+		} else {
+			logutil.Logger(ctx).Warn("can not retry txn",
+				zap.String("label", s.getSQLLabel()),
+				zap.Error(err),
+				zap.Bool("isBatchInsert", s.sessionVars.BatchInsert),
+				zap.Bool("isPessimistic", isPessimistic),
+				zap.Int64("commitRetryLimit", commitRetryLimit))
```

Comment:
Plz add the value of `tidb_disable_txn_auto_retry` and `tidb_retry_limit` and `InRestrictedSQL`. `commitRetryLimit` is not very useful for debug.

Reply:
@jackysp PTAL